src/kernel/mok_user_reply/mok_user_reply.rs

```diff
@@ -0,0 +1,62 @@
+use std::{sync::{Arc, Mutex}, thread};
+use serde::{de::DeserializeOwned, Serialize};
+
+use crate::kernel::{link::Link, str_err::str_err::StrErr};
+use super::query_struct::QueryStruct;
+///
+/// Struct to imitate user's answer's
+pub struct MokUserReply {
+    /// where store info about recieve and sender channel's
+    link: Link,
+    /// value to stop thread that await request's
+    stop_signal: Arc<Mutex<bool>>,
+}
+//
+//
+impl MokUserReply {
+    ///
+    /// Struct constructor
+    pub fn new(link: Link) -> Self {
+        Self { 
+            link,
+            stop_signal: Arc::new(Mutex::new(false)), 
+        }
+    }
+    ///
+    /// Running user service
+    pub fn run(self) -> Result<thread::JoinHandle<()>, String> {
```

Comment:
Move it to the [Service](https://github.com/a-givertzman/rust-sal-sync/blob/master/src/services/service/service.rs) impl
use [this as example](https://github.com/a-givertzman/cma-server/blob/master/src/services/task/task.rs )